src/Features/ERDDAP/Platform/Observations/CurrentConditions/wind.tsx

```diff
@@ -2,80 +2,177 @@
+  }
+
+  if (timeSeries.length < 0) {
+    return <OtherWindCard platform={platform}>No wind data</OtherWindCard>
+  }
+
+  return <LoadWindCard platform={platform} timeSeries={timeSeries} />
+}
+
+interface LoadWindCardProps extends WindCardProps {
+  timeSeries: PlatformTimeSeries[]
+}
+
+/**
+ * Load data for wind time series, and connect currently selected unit_system
+ */
+const LoadWindCard: React.FC<LoadWindCardProps> = ({ platform, timeSeries }) => {
+  const unit_system = useUnitSystem()
+  const { isLoading, data } = useDatasets(timeSeries)
+
+  if (isLoading) {
+    return <OtherWindCard platform={platform}>Loading wind data</OtherWindCard>
+  }
+
+  if (data) {
+    const aDayAgo = new Date()
+    aDayAgo.setDate(aDayAgo.getDate() - 1)
+
+    const filteredDatasets = (data as DataTimeSeries[])
+      .map((ds) => ({
+        ...ds,
+        timeSeries: ds.timeSeries.filter((r) => aDayAgo < r.time),
+      }))
+      .filter((ds) => ds.timeSeries.length > 0)
+
+    if (filteredDatasets.length > 0) {
+      return (
+        <DisplayWindCard
+          platform={platform}
+          timeSeries={timeSeries}
+          datasets={filteredDatasets}
+          unit_system={unit_system}
+        />
+      )
+    }
   }
 
-  const data: DataTimeSeries[] = datasets.map((dataset) => ({
-    name: dataset.data_type.long_name,
-    timeSeries: dataset.readings,
-    unit: dataset.data_type.units,
-  }))
+  return <OtherWindCard platform={platform}>Error loading wind data.</OtherWindCard>
+}
+
+interface DisplayWindCardProps extends LoadWindCardProps {
+  datasets: DataTimeSeries[]
+  unit_system: UnitSystem
+}
 
-  const speed = datasets.filter((dataset) => dataset.data_type.standard_name.includes("speed"))
-  const gust = datasets.filter((dataset) => dataset.data_type.standard_name.includes("gust"))
-  const direction = datasets.filter((dataset) => dataset.data_type.standard_name.includes("direction"))
+/**
+ * Display wind datasets on a current conditions card
+ */
+export const DisplayWindCard: React.FC<DisplayWindCardProps> = ({ platform, datasets, unit_system }) => {
+  const { speed, gust, direction } = pickWindDatasets(platform, datasets)
+  const { speed: speedTimeSeries, gust: gustTimeSeries } = pickWindTimeSeries(platform)
 
-  const speedAndGust = [...speed, ...gust]
-  if (speedAndGust.length < 1) {
+  if ((!speed && !gust) || !(speedTimeSeries || gustTimeSeries)) {
     return null
   }
 
-  const data_converter = converter(speedAndGust[0].data_type.standard_name)
+  const data_converter = converter(speedTimeSeries!.data_type.standard_name)
 
   let speedTitle: string = ""
-  if (speed.length > 0 && speed[0].readings.length > 0) {
+  if (speed) {
     speedTitle =
       " - " +
-      round(data_converter.convertToNumber(speed[0].readings[speed[0].readings.length - 1].reading, unit_system), 1) +
+      round(data_converter.convertToNumber(speed.timeSeries[speed.timeSeries.length - 1].reading, unit_system), 1) +
```

Comment:
Codacy found an issue: [Identifier 'data_converter' is not in camel case.](https://app.codacy.com/manual/gmri/Neracoos-1-Buoy-App/pullRequest?prid=6166252)